src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageService.TemporaryStorageStreamHandle.cs

```diff
@@ -20,9 +23,13 @@ Stream ITemporaryStorageStreamHandle.ReadFromTemporaryStorage(CancellationToken
 
         public UnmanagedMemoryStream ReadFromTemporaryStorage(CancellationToken cancellationToken)
         {
-            var storage = new TemporaryStreamStorage(
-                storageService, memoryMappedFile, this.Identifier.Name, this.Identifier.Offset, this.Identifier.Size);
-            return storage.ReadStream(cancellationToken);
+            using (Logger.LogBlock(FunctionId.TemporaryStorageServiceFactory_ReadStream, cancellationToken))
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var info = new MemoryMappedInfo(memoryMappedFile, Identifier.Name, Identifier.Offset, Identifier.Size);
+                return info.CreateReadableStream();
+            }
```

Comment:
this is just the old storage.ReadStream method inlined (so we could delete TemporaryStreamStorage).